src/main/java/org/zalando/nakadi/exceptions/NakadiRuntimeException.java

```diff
@@ -0,0 +1,17 @@
+package org.zalando.nakadi.exceptions;
+
+/* This exception is meant to be caught and unwrap the real excpetion */
+public class NakadiRuntimeException extends RuntimeException {
+
+    private final Exception exception;
+
+    public NakadiRuntimeException(final Exception exception) {
+        super(exception);
+        this.exception = exception;
+    }
+
+    public Exception getException() {
```

Comment:
There are logic in CleanupState which expect Exception.
And I just rename ExceptionWrapper to NakadiRuntimeException. And port Scala Try to be more functioal :)